### PR TITLE
Spam protection: Added ability to enable/disable email notifications

### PR DESCRIPTION
**Pull request by vndpal**
Spam Protection: Ability to pause/resume email notifications #128

This PR offers our users the ability to pause or resume email notifications for document views.
Users can enable or disable this option by editing a link.

image

By default, this setting will be enabled, which means that document view emails will be sent to them. If they want to stop these emails, they can modify the link and turn off the option. Also, when creating a new link, they have the choice to set the notification option to "disabled."

### DB Migration

Steps I followed to add new column and prisma migration

1. Modified the schema.prisma file to include the new column enableNotification
2. Ran this command `npx prisma migrate diff --from-url <DATABASE_URL_PROD> --to-schema-datamodel prisma/schema.prisma --script > prisma/migrations/202310311254_NewColumnEnableNotificationLinkTable/migration.sql` to create migration.sql file.
3. Ran this command `npx prisma db execute --url <DATABASE_URL_PROD> --file prisma/migrations/202310311254_NewColumnEnableNotificationLinkTable/migration.sql` to push the schema changes to the database
4. Then ran npx prisma generate to update the prisma-client

### Email Screen:

I have added a text message at the bottom of the email which let our users know about the option to disable the email notification for the link

image

Let me know if this text needs to be changed or removed completely.

### CODE REVIEW

File: components/emails/viewed-document.tsx [hunk 1]
```diff
@@ -77,6 +77,10 @@ export default function ViewedDocument({
                 If you have any feedback or questions about this email, simply
                 reply to it. I&apos;d love to hear from you!
               </Text>
+              <Text className="text-xs">
+                To stop email notifications for this link views, edit the link
```
Comment (mfts) [inline components/emails/viewed-document.tsx]:
```suggestion
                To stop email notifications for this link, edit the link
```

**Review by mfts — CHANGES_REQUESTED** [review]
@vndpal great job! minor text change. 

Otherwise, please update from `main` and resolve the 3 conflicts (nothing difficult). 
Happy to merge it afterwards :)

File: components/emails/viewed-document.tsx (hunk 1 above)
Comment (vndpal) [reply components/emails/viewed-document.tsx]: text changes done as suggested

**Comment by vndpal** [comment]
@mfts resolved conflicts and made minor text changes as requested.

**Review by mfts — APPROVED** [review]
LGTM 🚀

**Pull request merged**